client/web/src/Structure/Topics/Components/startQuizModal.jsx

```diff
@@ -21,9 +22,9 @@ const StartQuizModal = ({ open, handleClose, topic, topicId }) => {
   const isSuccessQuiz = useSelector((state) => state.topics.isSuccessQuiz);
   const startQuiz = () => {
     dispatch(topicActions.attemptQuiz(topicId)).then((res) => {
-      if (res === "success") {
+      if (res.final === "success") {
         history.push("/quiz_questions");
-      } else {
+      } else if (res.final === "failure") {
         alert("Unable to start Quiz try later");
```

Comment:
why dont you just return res.final as Boolean

Reply:
I always use success and failure for easy understanding

Reply:
ok